user_sync/app.py

```diff
@@ -105,11 +105,16 @@ def init_log(logging_config):
     }
     
     console_log_level = level_lookup.get(options['console_log_level'])
-    if (console_log_level != None):
-        console_log_handler.setLevel(console_log_level)
-    
+    if (console_log_level == None):
+        console_log_level = logging.DEBUG
+        logger.log(logging.INFO, 'Unknown console log level: %s setting to debug' % options['console_log_level'])
+    console_log_handler.setLevel(console_log_level)
+
     if options['log_to_file'] == True:
-        file_log_level = level_lookup.get(options['file_log_level'], logging.NOTSET)
+        file_log_level = level_lookup.get(options['file_log_level'])
+        if (file_log_level == None):
+            file_log_level = logging.DEBUG
+            logger.log(logging.INFO, 'Unknown file log level: %s setting to debug' % options['file_log_level'])
```

Comment:
@phil-levy - I have updated the logger.log to warning.

I think it is ok as the file log level is defaulted with this line in the init:

builder.set_string_value('file_log_level', 'debug')

* If file_log_level is not specified, then we will use the default specified in the init and we won’t output the message
* If file log_level is set to an invalid value e.g. TEST, then none will be returned from the .get, we will set the level to debug and output the message
* If file log_level is set to a valid value, then we don’t go into the if, we use the valid file_log_level.

This would be an issue however with the console_log_level as the default set for that in the init is None, therefore if it was missing from the config file you would get the error message and have the value defaulted. Changing the default value to info (in response to your other comment) will address this I believe.